packages/experiments/src/components/Button/Button.styles.ts

```diff
@@ -0,0 +1,232 @@
+          borderRadius: '50%',
+          contentPadding: ''
+        }
+      }
+    },
+    styleVariables
+  );
+
+  function getButtonStylesFromState(state: IButtonStyleVariablesTypes): Partial<IButtonStyles> {
+    if (state) {
+      return {
+        root: [
+          {
+            padding: 0,
+            backgroundColor: state.backgroundColor,
+            color: state.color,
+            borderColor: state.borderColor,
+            display: 'inline-block',
+            justifyContent: 'center',
+            boxSizing: 'border-box',
+            borderStyle: 'solid',
+            borderWidth: state.borderWidth,
+            borderRadius: state.borderRadius,
+            userSelect: 'none',
+            textDecoration: 'none',
+            textAlign: 'center',
+            verticalAlign: 'baseline',
+            overflow: 'hidden',
+            lineHeight: '1',
+            width: state.width,
+            height: state.height,
+            minWidth: state.minWidth,
+            minHeight: state.minHeight,
+
+            fontSize: state.textSize,
+            fontFamily: state.textFamily,
+            // tslint:disable-next-line:no-any
+            fontWeight: state.textWeight as any,
+
+            selectors: {
+              ':hover': {
+                backgroundColor: state.backgroundColorHovered,
+                color: state.colorHovered,
+                borderColor: state.borderColorHovered
+              },
+
+              ':hover:active': {
+                backgroundColor: state.backgroundColorPressed,
+                color: state.colorPressed,
+                borderColor: state.borderColorPressed
+              },
+              [`:hover .${globalClassNames.icon}`]: {
+                color: state.iconColorHovered
+              },
+              [`:hover:active .${globalClassNames.icon}`]: {
+                color: state.iconColorPressed
+              }
+            }
+          }
+        ],
+        icon: [
+          {
+            display: 'flex',
+            fontSize: state.iconSize,
+            color: state.iconColor,
+            fill: state.iconColor,
+            // tslint:disable-next-line:no-any
+            fontWeight: state.iconWeight as any
+          },
+          globalClassNames.icon
+        ],
+        stack: {
+          padding: state.contentPadding,
+          height: '100%'
+        },
+        text: {
+          overflow: 'visible'
+        }
+      };
+    }
+
+    // no state
+    return {};
+  }
+
+  function getButtonStylesFromVariant(variantVariables: { [PState in IButtonStates]: IButtonStyleVariablesTypes }): Partial<IButtonStyles> {
+    if (variantVariables) {
+      return concatStyleSets(
+        getButtonStylesFromState(variantVariables.baseState),
+        !disabled && getButtonStylesFromState(variantVariables.enabled),
+        !disabled && expanded && getButtonStylesFromState(variantVariables.expanded),
+        disabled && getButtonStylesFromState(variantVariables.disabled)
```

Comment:
Do you think this will become a generalized pattern we can centralize in Foundation?

Reply:
definitely will be refactoring.